wear/src/main/java/io/homeassistant/companion/android/tiles/ThermostatTile.kt

```diff
@@ -0,0 +1,303 @@
+                    thermostatTileDao.add(updated)
+                }
+
+                tile.setTileTimeline(
+                    if (serverManager.isRegistered()) {
+                        timeline(
+                            tileConfig,
+                            targetTemp
+                        )
+                    } else {
+                        loggedOutTimeline(
+                            this@ThermostatTile,
+                            requestParams,
+                            R.string.thermostat,
+                            R.string.thermostat_tile_log_in
+                        )
+                    }
+                ).build()
+            } catch (e: Exception) {
+                Log.e(TAG, "Unable to fetch entity ${tileConfig?.entityId}", e)
+
+                tile.setTileTimeline(
+                    primaryLayoutTimeline(
+                        this@ThermostatTile,
+                        requestParams,
+                        null,
+                        R.string.tile_fetch_entity_error,
+                        R.string.refresh,
+                        ActionBuilders.LoadAction.Builder().build()
+                    )
+                ).build()
+            }
+        }
+    }
+
+    override fun onTileResourcesRequest(requestParams: ResourcesRequest): ListenableFuture<Resources> = serviceScope.future {
+        Resources.Builder()
+            .setVersion(requestParams.version)
+            .addIdToImageMapping(
+                RESOURCE_REFRESH,
+                ResourceBuilders.ImageResource.Builder()
+                    .setAndroidResourceByResId(
+                        ResourceBuilders.AndroidImageResourceByResId.Builder()
+                            .setResourceId(io.homeassistant.companion.android.R.drawable.ic_refresh)
+                            .build()
+                    ).build()
+            )
+            .build()
+    }
+
+    override fun onTileAddEvent(requestParams: EventBuilders.TileAddEvent) = runBlocking {
+        withContext(Dispatchers.IO) {
+            val dao = AppDatabase.getInstance(this@ThermostatTile).thermostatTileDao()
+            if (dao.get(requestParams.tileId) == null) {
+                dao.add(ThermostatTile(id = requestParams.tileId))
+            } // else already existing, don't overwrite existing tile data
+        }
+    }
+
+    override fun onTileRemoveEvent(requestParams: EventBuilders.TileRemoveEvent) = runBlocking {
+        withContext(Dispatchers.IO) {
+            AppDatabase.getInstance(this@ThermostatTile)
+                .thermostatTileDao()
+                .delete(requestParams.tileId)
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        serviceScope.cancel()
+    }
+
+    private suspend fun timeline(tileConfig: ThermostatTile?, targetTemperature: Float): Timeline = Timeline.fromLayoutElement(
```

Comment:
Isn't `tileConfig` always non-null at this point (so `tileConfig: ThermostatTile` could be used)?